azurelinuxagent/common/protocol/util.py

```diff
@@ -254,13 +305,14 @@ def _save_protocol(self, protocol_name):
         protocol_file_path = self._get_protocol_file_path()
         try:
             fileutil.write_file(protocol_file_path, protocol_name)
-        except IOError as e:
+        except (IOError, OSError) as e:
             logger.error("Failed to save protocol endpoint: {0}", e)
 
     def clear_protocol(self):
         """
-        Cleanup previous saved endpoint.
+        Cleanup previous saved protocol endpoint.
         """
+        self._clear_wireserver_endpoint()
         logger.info("Clean protocol")
```

Comment:
Is this log needed here now, with the `logger.info("Clean wireserver endpoint")` log added in `self._clear_wireserver_endpoint()`?

Reply:
Since nothing is expected to call _clear_wireserver_endpoint() directly, I'll make a single log entry stating both variables are getting cleared. Thank you for highlighting the excessive log entries. :)